src/vs/workbench/contrib/terminal/browser/terminalView.ts

```diff
@@ -76,6 +84,15 @@ export class TerminalViewPane extends ViewPane {
 				this.layoutBody(this._parentDomElement.offsetHeight, this._parentDomElement.offsetWidth);
 			}
 		});
+
+		this._dropdownMenu = this._menuService.createMenu(MenuId.TerminalToolbarContext, this._contextKeyService);
+
+		this._terminalService.onRequestAvailableProfiles(async () => {
+			if (this._tabButtons) {
+				await this._updateTabActionBar();
+			}
+		});
```

Comment:
That event is only meant for `MainThreadTerminalService`, I think this is what we're after:

```
this._terminalService.getAvailableProfilesAsync().then(profiles => ...)
this._terminalService.onProfilesConfigChanged(profiles => { ... })
```

I think we should also pass the profiles into the `_onProfilesConfigChanged` event in terminal service and handle that inside the dropdown as well, that's probably what was causing it to update every 10 seconds.

Reply:
what about this? so we get the profiles when they're ready for the first time and only fetch them again if the config has changed or if the window is reloaded?
```

		this._terminalService.onRequestAvailableProfiles(() => {
			if (!this._requestedAvailableProfiles) {
				this._terminalService.getAvailableProfilesAsync().then(profiles => {
					if (this._tabButtons) {
						this._updateTabActionBar(profiles);
					}
				});
				this._requestedAvailableProfiles = true;
			}
		});
		this._terminalService.onProfilesConfigChanged(() => {
			this._terminalService.getAvailableProfilesAsync().then(profiles => {
				if (this._tabButtons) {
					this._updateTabActionBar(profiles);
				}
			});
		});
```

Reply:
Oh we want to trigger updates as well, forgot that this replaces the dropdown's updating. Still I don't think `this._terminalService.onRequestAvailableProfiles` is correct though and should be `onProfilesConfigChanged`?

Reply:
found an issue w this - if you close all of the terminals and reopen the panel, it doesn't repopulate the list since config hasn't changed and since it's already been set to true

Reply:
for now have fixed this by when the pane gets shown, call profilesConfigchanged

Reply:
`onProfilesConfigChanged` isn't fired though unless the settings are changed

Reply:
Created https://github.com/microsoft/vscode/issues/121812